File: android/app/src/main/java/com/costumerental/app/models/Reservation.java
package com.costumerental.app.models;

// Reservation model
public class Reservation {
    private int id;
    private int costume_id;
    private int user_id;
    private String start_date;
    private String end_date;
    private String status; // pending, approved, rejected
    private Costume costume;
    private User user;

    public Reservation() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getCostume_id() {
        return costume_id;
    }

    public void setCostume_id(int costume_id) {
        this.costume_id = costume_id;
    }

    public int getUser_id() {
        return user_id;
    }

    public void setUser_id(int user_id) {
        this.user_id = user_id;
    }

    public String getStart_date() {
        return start_date;
    }

    public void setStart_date(String start_date) {
        this.start_date = start_date;
    }

    public String getEnd_date() {
        return end_date;
    }

    public void setEnd_date(String end_date) {
        this.end_date = end_date;
    }

    public Costume getCostume() {
        return costume;
    }

    public void setCostume(Costume costume) {
        this.costume = costume;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}

File: backend/app/Models/Reservation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Reservation extends Model
{
    use HasFactory;

    protected $fillable = [
        'costume_id',
        'user_id',
        'start_date',
        'end_date',
        'status',
    ];

    // Get the costume
    public function costume()
    {
        return $this->belongsTo(Costume::class);
    }

    // Get the user
    public function user()
    {
        return $this->belongsTo(User::class);
    }
}
